### PR TITLE
Finish package manager abstraction cleanup

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge consists of some final clean up for the package manager abstraction project. Here we complete updating package definitions, documentation, tutorials, and updating tests to make sure they pass.

### CODE REVIEW

File: lib/ramble/ramble/test/end_to_end/spack_env_cache.py [hunk 1]
```diff
@@ -70,50 +74,53 @@ def test_spack_env_cache():
...
+            content = f.read()
+            assert "spack install" not in content
+            assert "spack concretize" not in content
+
+        test4_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_gmx50.test4.out")
+        with open(test4_log, "r") as f:
+            content = f.read()
+            assert "spack install" not in content
+            assert "spack concretize" not in content
+    except RunnerError as e:
```
Comment (linsword13) [inline lib/ramble/ramble/test/end_to_end/spack_env_cache.py:125]: Curious why this is needed now?

File: lib/ramble/docs/configuration_files.rst [hunk 2]
```diff
@@ -624,4 +626,27 @@ format is as follows:
...
+
+.. _variants-config:
+
+----------------
+Variants Section
+----------------
+
+The variants config section is used to customize variants to the experiment creation.
+These can include application defined variants, or higher level Ramble provided
+variants. Currently, the only supported variants is ``package_manager`` which
```
Comment (linsword13) [inline lib/ramble/docs/configuration_files.rst:642]: What's the default behavior currently? Does it default to use spack if no variant config is set?

File: lib/ramble/ramble/test/end_to_end/spack_env_cache.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/test/end_to_end/spack_env_cache.py:125]: It is not strictly needed in this PR, I just noticed it when I was going through the unit tests.

I believe if you take develop today on a machine that doesn't have spack installed, this unit test will fail. This change allows it to be skipped instead.

File: lib/ramble/docs/configuration_files.rst (hunk 2 above)
Comment (douglasjacobsen) [reply lib/ramble/docs/configuration_files.rst:642]: Ah, good point. Let me add that into the documentation.

**Pull request merged**